pilot/pkg/bootstrap/certcontroller.go

```diff
@@ -279,3 +305,27 @@ func (s *Server) loadIstiodCert() error {
 	s.certMu.Unlock()
 	return nil
 }
+
+func (s *Server) getRootCertFromMeshConfig(signerName string) ([]byte, error) {
+	meshConfig := s.environment.Mesh()
+	caCertificates := meshConfig.CaCertificates
+	if len(caCertificates) == 0 {
+		return nil, fmt.Errorf("no caCertificates defined in mesh config")
+	}
+	for _, caCertificate := range caCertificates {
+		signers := caCertificate.CertSigners
+		if len(signers) == 0 {
+			return nil, fmt.Errorf("no cert signers defined in mesh config")
+		}
+		sort.Strings(signers)
+		index := sort.SearchStrings(signers, signerName)
```

Comment:
Its faster + simpler to just iterate over the list and find it

Reply:
Thanks! Fixed